code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -118,33 +118,45 @@
 	. = ..()
 	if(!up)
 		. += visor_icon
-
-/obj/item/clothing/head/helmet/space/plasmaman/attackby(obj/item/hitting_item, mob/living/user)
-	. = ..()
-	if(istype(hitting_item, /obj/item/toy/crayon))
-		if(smile == FALSE)
-			var/obj/item/toy/crayon/CR = hitting_item
-			to_chat(user, span_notice("You start drawing a smiley face on the helmet's visor.."))
-			if(do_after(user, 2.5 SECONDS, target = src))
-				smile = TRUE
-				smile_color = CR.paint_color
-				to_chat(user, "You draw a smiley on the helmet visor.")
-				update_appearance()
-		else
+	if(smile)
+		var/mutable_appearance/slimey = mutable_appearance(icon, smile_state)
+		slimey.color = smile_color
+		. += slimey
+
+/obj/item/clothing/head/helmet/space/plasmaman/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(istype(tool, /obj/item/toy/crayon))
+		if(smile)
 			to_chat(user, span_warning("Seems like someone already drew something on this helmet's visor!"))
-		return
-	if(istype(hitting_item, /obj/item/clothing/head))
-		var/obj/item/clothing/hitting_clothing = hitting_item
-		if(hitting_clothing.clothing_flags & STACKABLE_HELMET_EXEMPT)
-			to_chat(user, span_notice("You cannot place [hitting_clothing.name] on helmet!"))
-			return
-		if(attached_hat)
-			to_chat(user, span_notice("There's already something placed on helmet!"))
-			return
-		attached_hat = hitting_clothing
-		to_chat(user, span_notice("You placed [hitting_clothing.name] on helmet!"))
-		hitting_clothing.forceMove(src)
+			return ITEM_INTERACT_BLOCKING
+
+		var/obj/item/toy/crayon/crayon = tool
+		to_chat(user, span_notice("You start drawing a smiley face on the helmet's visor..."))
+		if(!do_after(user, 2.5 SECONDS, target = src))
+			return ITEM_INTERACT_BLOCKING
+
+		smile = TRUE
+		smile_color = crayon.paint_color
+		to_chat(user, "You draw a smiley on the helmet visor.")
 		update_appearance()
+		return ITEM_INTERACT_SUCCESS
+
+	if(!istype(tool, /obj/item/clothing/head))
+		return NONE
+
+	var/obj/item/clothing/hitting_clothing = tool
+	if(hitting_clothing.clothing_flags & STACKABLE_HELMET_EXEMPT)
+		to_chat(user, span_notice("You cannot place [hitting_clothing.name] on helmet!"))
```

Comment:
I know you didn't edit this but ``on helmet`` sucks so much can you please replace it with src or something

Reply:
some of plasmeme helmet names are 6 words long, I feel like I should just fix up the grammar instead of [src]ing every single one of these tbh